build.sbt

```diff
@@ -90,12 +14,22 @@ val newerCompilerLintSwitches = Seq(
   "-Ywarn-numeric-widen" // In 2.10 this produces a some strange spurious error
 )
 
-scalacOptions ++= allVersionCompilerLintSwitches
-
-scalacOptions in (Compile, console) ~= (_ filterNot (Set("-Xfatal-warnings", "-Ywarn-unused-import").contains))
-
-scalacOptions in (Test, console) := (scalacOptions in (Compile, console)).value
-
-scalacOptions ++= PartialFunction.condOpt(CrossVersion.partialVersion(scalaVersion.value)){
+lazy val settings = Seq(
+  scalaVersion := "2.12.0",
+  crossScalaVersions := Seq("2.10.5", "2.11.8", "2.12.0"),
+  scalacOptions ++= Seq("-feature"),
+  scalacOptions ++= allVersionCompilerLintSwitches,
+  scalacOptions in (Compile, console) ~= (_ filterNot (Set("-Xfatal-warnings", "-Ywarn-unused-import").contains)),
+  scalacOptions in (Test, console) := (scalacOptions in (Compile, console)).value,
+  scalacOptions ++= PartialFunction.condOpt(CrossVersion.partialVersion(scalaVersion.value)){
     case Some((2, scalaMajor)) if scalaMajor >= 11 => newerCompilerLintSwitches
-}.toList.flatten
+  }.toList.flatten,
+  initialize := {
+    val required = "1.8"
+    val current  = sys.props("java.specification.version")
+    assert(current == required, s"Unsupported JDK: java.specification.version $current != $required")
+  }
+)
+
+lazy val core = (project in file("core")).
```

Comment:
Suggestion: Move this to the top of the file to set the precedent that the project are listed up there.